blaze-client/src/main/scala/org/http4s/blaze/client/BlazeClient.scala

```diff
@@ -66,95 +63,82 @@ object BlazeClient {
       requestTimeout: Duration,
       scheduler: TickWheelExecutor,
       ec: ExecutionContext,
-  )(implicit F: ConcurrentEffect[F]) =
-    Client[F] { req =>
-      Resource.suspend {
-        val key = RequestKey.fromRequest(req)
+  )(implicit F: ConcurrentEffect[F]): Client[F] =
+    new BlazeClient[F, A](manager, responseHeaderTimeout, requestTimeout, scheduler, ec)
+}
 
-        // If we can't invalidate a connection, it shouldn't tank the subsequent operation,
-        // but it should be noisy.
-        def invalidate(connection: A): F[Unit] =
-          manager
-            .invalidate(connection)
-            .handleError(e => logger.error(e)(s"Error invalidating connection for $key"))
+private class BlazeClient[F[_], A <: BlazeConnection[F]](
+    manager: ConnectionManager[F, A],
+    responseHeaderTimeout: Duration,
+    requestTimeout: Duration,
+    scheduler: TickWheelExecutor,
+    ec: ExecutionContext,
+)(implicit F: ConcurrentEffect[F])
+    extends DefaultClient[F] {
 
-        def borrow: Resource[F, manager.NextConnection] =
-          Resource.makeCase(manager.borrow(key)) {
-            case (_, ExitCase.Completed) =>
-              F.unit
-            case (next, ExitCase.Error(_) | ExitCase.Canceled) =>
-              invalidate(next.connection)
-          }
+  override def run(req: Request[F]): Resource[F, Response[F]] = for {
+    _ <- Resource.pure[F, Unit](())
```

Comment:
Why this rather than 
```
{
  val key = RequestKey.fromRequest(req)
  for {
   ...
   ```

Reply:
There's no reason other than a stylistic choice. It results in "flatter" code. I guess that's something that is popular in my team, and looks strage to everyone outside of the team ;).  I'll change it to the `val key` version.